### PR TITLE
IA Project: add post page creation sources tracking

### LINKED ISSUE #10744: Add tracking of where we create page/post from in the app
In the Information Architecture project we are adding a bottom sheet from where it is possible to create a new post/page.

As part of this task we want to add event tracking to trace from where the user creates a new post/page.

Currently we have the `editor_post_created` event that has the `created_post_source` property in it. This property traces the following categories:

1. post-list
2. shared-from-external-app
3. media-library
4. quick-press

We want to detail the post-list category more in detail.

### PR DESCRIPTION
**Pull request by develric**
Fixes #10744 

To better detail the post-list tracking category we created the following sub categories:

```
POST_FROM_SHORTCUT: post created from android shortcut long pressing on app icon
POST_FROM_POSTS_LIST: post created from the posts list fab
PAGE_FROM_PAGES_LIST: page created from the pages list fab
POST_FROM_NAV_BAR: post created from the navigation bar (until we will remove it)
POST_FROM_MY_SITE: post created from bottom sheet in my site screen
PAGE_FROM_MY_SITE: page created from bottom sheet in my site screen
POST_FROM_STATS: post created from stats empty view when no stats/post available yet
POST_FROM_NOTIFS_EMPTY_VIEW: post created from notifications unread page when empty
NO_DETAIL: all other cases container
```
Please note that currently `POST_FROM_MY_SITE` and `PAGE_FROM_MY_SITE` events are only available in zalpha flavor where the INFORMATION_ARCHITECTURE_AVAILABLE feature flag is activated, whereas the `POST_FROM_NAV_BAR` is available only in the vanilla (or wasabi); so to get all the above events you will need to use both the vanilla and the zalpha flavors.

## To test
1. Be sure that tracking is enabled in Me > App Settings > Privacy settings > Collect information ( 😁 )
2. Launch the app and trigger the above events
3. Check the logcat to see that you get the `editor_post_created` event with the `created_post_source_detail` property value like in the following example:

```
Tracked: editor_post_created, Properties: {"blog_id":157783356,"created_post_source_detail":"other","post_type":"post-from-my-site","created_post_source":"post-list","is_jetpack":false}
```

NOTE: if merged into develop this PR adds the tracking capability also without the feature flag activated. cc @osullivanchris 

PR submission checklist:

- [x] I have considered adding unit tests where possible.

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/PagePostCreationSources.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package org.wordpress.android.ui
+
+enum class PagePostCreationSources(val label: String) {
+    // post created from android shortcut long pressing on app icon
+    POST_FROM_SHORTCUT("post-from-short-cut"),
```
Comment (planarvoid) [inline WordPress/src/main/java/org/wordpress/android/ui/PagePostCreationSources.kt]: I think it should be `post-from-shortcut`

File: WordPress/src/main/java/org/wordpress/android/ui/PagePostCreationSources.kt [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    // post created from the posts list fab
+    POST_FROM_LIST("post-from-list"),
+    // page created from the pages list fab
+    PAGE_FROM_LIST("page-from-list"),
```
Comment (planarvoid) [inline WordPress/src/main/java/org/wordpress/android/ui/PagePostCreationSources.kt]: I'm not sure these two are self-explanatory enough. What do you think about something like `page-from-pages-screen`?

**Comment by develric** [comment]
Hi @planarvoid , thanks for reviewing this 🙇‍♂️. 

> I was wondering if it might be confusing that we have both created_post_source and created_post_source_detail.

I was thinking about this, actually what we are doing here is detailing the `post-list `case of the  `created_post_source` but I did not want to remove that in case it is used in some analysis (I checked the funnels in tracks and AFAIU it is not directly used but I'm not fully sure how to assert that we do not use in some other place as well; also the `post-list` cumulative information seems a pretty useful consolidated data in itself in my understanding). 

Wdyt if we change the naming of the `PagePostCreationSources` to something like `PostListCreationSourceDetail` and `created_post_source_detail` to `post_list_create_source_detail`? (Keeping it empty when we are not in the `post-list` case)

**Comment by planarvoid** [comment]
> I was thinking about this, actually what we are doing here is detailing the post-list case of the created_post_source but I did not want to remove that in case it is used in some analysis (I checked the funnels in tracks and AFAIU it is not directly used but I'm not fully sure how to assert that we do not use in some other place as well; also the post-list cumulative information seems a pretty useful consolidated data in itself in my understanding).

I think that makes sense. What do you think if the new field contained the previous sources (like "shared-from-external-app") as well and we would call it something like `detailed_created_post_source`. `created_post_source` would basically be a subset of `detailed_created_post_source` but anyone can use the detailed source instead of the old one. We'd keep the old field for consistency.

File: WordPress/src/main/java/org/wordpress/android/ui/PagePostCreationSources.kt (hunk 1 above)
Comment (develric) [reply WordPress/src/main/java/org/wordpress/android/ui/PagePostCreationSources.kt]: Correct...sorry for the typo 😊

File: WordPress/src/main/java/org/wordpress/android/ui/PagePostCreationSources.kt (hunk 2 above)
Comment (develric) [reply WordPress/src/main/java/org/wordpress/android/ui/PagePostCreationSources.kt]: Makes sense...changed here 562213bef2f17d2322bc059bb2e00022122849b7

**Comment by develric** [comment]
Hi @planarvoid , thanks for the additional chat on slack! Trying to report what we decided, please  add/integrate if I missed or misunderstood something, thanks 😊.

We will use the `created_post_source_detail`property to further detail the already available `created_post_source` property. Currently only the `post-list` category will be further detailed, all the other cases will fall under the `no-detail` container. In the future we can add other detailed category for `post-list` or the other `created_post_source` elements.

**Pull request merged**